common/roles/cert-perms/tasks/main.yml

```diff
@@ -1,111 +1,104 @@
-      - '/etc/letsencrypt/live/{{ cert_dir }}'
-  loop_control:
-    loop_var: path
-
-- name: Set the unix mode for the directory accessible by root only
-  tags: cert
-  file:
-    path: '{{ path }}'
-    owner: root
-    group: root
-    mode: '0700'
-    recurse: false
   with_items:
-      - '/etc/letsencrypt/archive/{{ cert_dir }}'
-      - '/etc/letsencrypt/live/{{ cert_dir }}'
-  loop_control:
-    loop_var: path
-
-
-# First, set the acl for the files in the directory. This should cover the case
-# when the files have been synchronised
-- name: Set the acl permissions for all the certificate files in the directory
-  tags: cert
-  acl:
-    path: '{{ path }}'
-    entity: '{{ entity_group }}'
-    etype: user
-    permissions: 'r--'
-    recursive: true
-    state: present
-  with_items:
-      - '/etc/letsencrypt/archive/{{ cert_dir }}'
-      - '/etc/letsencrypt/live/{{ cert_dir }}'
+    - '{{ certs.files | selectattr("mode", "ne", "0600") | list }}'
+    - '{{ keys.files | selectattr("mode", "ne", "0600") | list }}'
   loop_control:
-    loop_var: path
+    loop_var: file
 
-- name: Set the acl permissions for the directory itself
+# Now, we set the default acl for this directory: new certificates
+# will be automatically marked as readable by the entity.
+# This doesn't need to be done for nginx, as the certificates
+# are read by root anyway.
+- name: Set the acl and default acl permissions for the directory
+  when: access_private_key
   tags: cert
   acl:
     path: '{{ path }}'
     entity: '{{ entity_group }}'
     etype: user
-    permissions: 'r-x'
+    permissions: 'r'
     recursive: false
     state: present
+    default: true
   with_items:
       - '/etc/letsencrypt/archive/{{ cert_dir }}'
-      - '/etc/letsencrypt/live/{{ cert_dir }}'
   loop_control:
     loop_var: path
 
-# Then, set the default permissions for the directory.
-# The new certificates created should have the right permissions
-- name: Set the default acl for the certificate directory iself
-  tags: cert
-  acl:
-    path: '{{ path }}'
-    entity: '{{ entity_group }}'
-    etype: user
-    recursive: false
-    default: yes
-    state: present
-    permissions: 'r--'
-  with_items:
-      - '/etc/letsencrypt/archive/{{ cert_dir }}'
-      - '/etc/letsencrypt/live/{{ cert_dir }}'
-  loop_control:
-    loop_var: path
+# Create the list of files to modify. When the entity is nginx,
+# do not include the private key
+- name: Create the list of files to modify
+  when: not access_private_key
+  set_fact:
+    files_list: '{{ certs.files }}'
 
-- name: Set the acl mask for the directory iself
+# In this case, this is a specific entity (e.g. openldap)
+# we include the private key too.
+- name: Create the list of files to modify
+  when: access_private_key
+  set_fact:
+    files_list: '{{ certs.files + keys.files }}'
+
+# Explicitly set the mask for this files as read. Because the parent
+# directory permissions are '0777', any ACL modification would create
```

Comment:
When are the directories `0777`?
I suppose it could be when setting rights from the `live/` symlink directories. Rights are no longer set through those directories. Is there another reason?

Reply:
I suppose I can try again with 750. I thought that was safe, as long as the files are not readable by everyone.